hashes/src/sha256.rs

```diff
@@ -989,7 +1011,7 @@ mod tests {
             0x8f, 0xf1, 0xf7, 0xa9, 0xd5, 0x69, 0x09, 0x59,
         ];
         let midstate_engine =
-            sha256::HashEngine::from_midstate(sha256::Midstate::from_byte_array(MIDSTATE), 64);
+            sha256::HashEngine::from_midstate(sha256::Midstate::new(MIDSTATE, 64), 64);
```

Comment:
Why does `from_midstate` still accept the length argument? It's already in the midstate.